### DISCUSSION TITLE
Install linkerd identity certificates with helm

### DISCUSSION THREAD

**Discussion by yaron-Kalatian** [linkerd/linkerd2#12574] [Help]
Hi , we use flux for helm charts . I want to deploy the linkerd-control-plane
I’m using linkerd2-edge for the linkerd-control-plane
it written:
certificates must live in a ConfigMap resource named linkerd-identity-trust-roots . Issuer certificates must live in a Secret named linkerd-identity-issuer
and then supply the files when running helm:

helm install linkerd-control-plane -n linkerd
–set-file identityTrustAnchorsPEM=ca.crt
–set-file identity.issuer.tls.crtPEM=issuer.crt
–set-file identity.issuer.tls.keyPEM=issuer.key
linkerd/linkerd-control-plane

I can create the ca.crt, issuer.crt ,issuer.key with step command , but how do I apply them to the CM linkerd-identity-trust-rootsand secret linkerd-identity-issuer to load them to git ?
Thanks

**Comment by adleong** [comment]
Hi @yaron-Kalatian 

If you're following the steps at https://linkerd.io/2.15/tasks/install-helm/ then the Helm installation will automatically create the necessary configmaps and secret resources.  I'm not sure I understand the part of your question about loading them into git, can you elaborate?

**Reply by yaron-Kalatian** [reply]
Hi @adleong  , we are using flux , so I'm not running the command: "helm install linkerd-control-plane -n linkerd ....."
I've download the values file of linkerd-contrl-plane  from artifacthub , and the flux should reconcile the code from git 
so how do I upload the ca.crt   issuer.crt issuer.key into ConfigMap resource named linkerd-identity-trust-roots. Issuer and a secret named linkerd-identity-issuer

https://artifacthub.io/packages/helm/linkerd2-edge/linkerd-control-plane/2024.4.5

the release.yaml from my gitlab : 
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: linkerd-control-plane
  namespace: linkerd
spec:
  interval: 1m
  chart:
    spec:
      chart: linkerd-control-plane
      version: '2024.4.5'
      sourceRef:
        kind: HelmRepository
        name: linkerd-edge
        namespace: flux-system
      interval: 1m
  values:
here I pasted the default values


 thanks  for your quick response
